### PR TITLE
fix: fix resolve error

### PR DESCRIPTION
**Pull request by iahu**
- resolve index file of all extensions
- add `jsx` and `tsx` scopes
- support typescript scope
- fix syntax error in resolve_as_directory

### CODE REVIEW

**Comment by iahu** [comment]
hi @aziz could you please take a look on my PR, thank you.

**Comment by niksy** [comment]
@aziz can we get this merged?

Pinging @cristianl since there was recent activity from him!

File: hyper_click/generic_path_resolver.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def resolve_as_file(self, path_name):
         if path.isfile(path_name):
             return path_name
 
-        with_ext = self.resolve_with_exts(path_name)
+        with_ext = self.resolve_with_exts(path_name) or self.resolve_with_exts(path.join(path_name, 'index'))
```
Comment (niksy) [inline hyper_click/generic_path_resolver.py]: @iahu is this necessary? There is already `index` file check in `resolve_index` method. Maybe that method should be changed to `self.resolve_with_exts(path.join(path_name, 'index'))` and this line left unmodified?

**Comment by cristianl** [comment]
Released as [2.3](https://github.com/aziz/SublimeHyperClick/releases/tag/2.3). Thank you @iahu!

**Comment by niksy** [comment]
@cristianl it’s not available on Package Control?

**Comment by jfcherng** [comment]
Probably should be tagged as `2.3.0` since `2.3` is not a valid semantic version.

https://packagecontrol.io/docs/submitting_a_package#Step_4


> 4. Pick a Versioning Strategy
>
> If you are using GitHub or BitBucket for your hosting, you will need to create a tag each time you want to make a new version available to users. The tag names must be a semantic version number.

**Pull request merged**